### PR TITLE
 Fix refresh of target if upstream no longer exists

### PR DESCRIPTION
**Pull request by onematchfox**
Resolves https://github.com/kevholditch/terraform-provider-kong/issues/69

### CODE REVIEW

File: kong/resource_kong_target_test.go [hunk 1]
```diff
@@ -160,12 +180,34 @@ func testAccCheckKongTargetDoesNotExist(targetResourceKey string, upstreamResour
...
 
-		targets, _ := testAccProvider.Meta().(*gokong.KongAdminClient).Targets().GetTargetsFromUpstreamId(rs.Primary.ID)
+		targets, err := testAccProvider.Meta().(*gokong.KongAdminClient).Targets().GetTargetsFromUpstreamId(rs.Primary.ID)
 
 		if len(targets) > 0 {
 			return fmt.Errorf("expecting zero target resources found %v", len(targets))
 		}
 
+		if err != nil {
+			return fmt.Errorf("Error thrown when trying to read target: %v", err)
```
Comment (kevholditch) [inline kong/resource_kong_target_test.go]: error string should begin with lowercase letter

File: kong/resource_kong_target_test.go [hunk 2]
```diff
@@ -160,12 +180,34 @@ func testAccCheckKongTargetDoesNotExist(targetResourceKey string, upstreamResour
...
+
+		if !ok {
+			return fmt.Errorf("not found: %s", upstreamResourceKey)
+		}
+
+		err := testAccProvider.Meta().(*gokong.KongAdminClient).Upstreams().DeleteById(rs.Primary.ID)
+
+		if err != nil {
+			return fmt.Errorf("could not delete kong upstream: %v", err)
+		}
```
Comment (kevholditch) [inline kong/resource_kong_target_test.go]: combine if and err declaration onto one line...
```
err := testAccProvider.Meta().(*gokong.KongAdminClient).Upstreams().DeleteById(rs.Primary.ID); if err != nil {
			return fmt.Errorf("could not delete kong upstream: %v", err)
		}
```

File: kong/resource_kong_target.go [hunk 3]
```diff
@@ -55,6 +55,14 @@ func resourceKongTargetCreate(d *schema.ResourceData, meta interface{}) error {
 func resourceKongTargetRead(d *schema.ResourceData, meta interface{}) error {
 
 	var ids = strings.Split(d.Id(), "/")
+
+	// First check if the upstream exists. If it does not then the target no longer exists either.
+	upstream, _ := meta.(*gokong.KongAdminClient).Upstreams().GetById(ids[0])
```
Comment (kevholditch) [inline kong/resource_kong_target.go]: combine if and variable declaration onto one line...
```
upstream, _ := meta.(*gokong.KongAdminClient).Upstreams().GetById(ids[0]); if upstream == nil ...
```

**Comment by onematchfox** [comment]
@kevholditch Thanks for the feedback. Changes have been made. Wasn't familiar with the combination of var initialization and check for result in Go prior to now so have started my day learning something :)

**Comment by kevholditch** [comment]
@onematchfox thanks for the contribution and helping to make the provider better.  Cool that you started the day learning something!  I'll merge your PR now and the changes will be in `v4.4.0` of the provider.

**Pull request merged**